bin/differential_tester_x86/LiftAndCompare.cpp

```diff
@@ -0,0 +1,541 @@
+#include <fenv.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+#include <llvm/ExecutionEngine/ExecutionEngine.h>
+#include <llvm/ExecutionEngine/GenericValue.h>
+#include <llvm/ExecutionEngine/Interpreter.h>
+#include <llvm/ExecutionEngine/MCJIT.h>
+#include <llvm/IR/Instructions.h>
+#include <llvm/IR/Verifier.h>
+#include <llvm/Passes/PassBuilder.h>
+#include <llvm/Support/DynamicLibrary.h>
+#include <llvm/Support/Endian.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/JSON.h>
+#include <llvm/Support/TargetSelect.h>
+#include <llvm/Transforms/Utils/Cloning.h>
+#include <remill/Arch/Arch.h>
+#include <remill/Arch/Name.h>
+#include <remill/Arch/X86/Runtime/State.h>
+#include <remill/BC/ABI.h>
+#include <remill/BC/IntrinsicTable.h>
+#include <remill/BC/Optimizer.h>
+#include <remill/BC/Util.h>
+#include <remill/OS/OS.h>
+#include <test_runner/TestRunner.h>
+
+#include <functional>
+#include <random>
+
+#include "Whitelist.h"
+#include "gtest/gtest.h"
+
+
+DEFINE_string(target_insn_file, "", "Path to input test cases");
+DEFINE_uint64(num_iterations, 2, "number of iterations per test case");
+DEFINE_string(repro_file, "", "File to output failing test cases");
+DEFINE_string(whitelist, "", "File listing instruction states not to check");
+DEFINE_bool(should_dump_functions, false, "Dump each function version");
+DEFINE_bool(stop_on_fail, false, "Stop on first failure");
+
+
+class DiffModule {
+ private:
+  std::unique_ptr<llvm::Module> mod;
+  llvm::Function *f1;
+  llvm::Function *f2;
+  std::string f1_insn_name;
+  std::string f2_insn_name;
```

Comment:
`std::vector` of tuples? Having 4 attrs like this is a lot of copy and paste - `GetF1` and `GetF2` can be easily unified as `GetF< N >` or `GetF( n )` - same for names.